deepchem/models/torch_models/gat.py

```diff
@@ -0,0 +1,225 @@
+    """
+    try:
+      from torch_geometric.nn import GATConv, global_mean_pool
+    except:
+      raise ValueError("This class requires PyTorch Geometric to be installed.")
+    super(GAT, self).__init__()
+    self.embedding = nn.Linear(in_node_dim, hidden_node_dim)
+    self.conv_layers = nn.ModuleList([
+        GATConv(
+            in_channels=hidden_node_dim,
+            out_channels=hidden_node_dim,
+            heads=heads,
+            concat=False,
+            dropout=dropout_rate) for _ in range(num_conv)
+    ])
+    self.pooling = global_mean_pool
+    self.fc = nn.Linear(hidden_node_dim, predicator_hidden_feats)
+    self.out = nn.Linear(predicator_hidden_feats, n_tasks)
+
+  def forward(self, data):
+    """Predict labels
+
+    Parameters
+    ----------
+    data: torch_geometric.data.Batch
+      A mini-batch graph data for PyTorch Geometric models.
+
+    Returns
+    -------
+    out: torch.Tensor
+      The output value, the shape is `(batch_size, n_out)`.
+    """
+    node_feat, edge_index = data.x, data.edge_index
+    node_feat = self.embedding(node_feat)
+
+    # convolutional layer
+    for conv in self.conv_layers:
+      node_feat = conv(node_feat, edge_index)
+
+    # pooling
+    graph_feat = self.pooling(node_feat, data.batch)
+    graph_feat = self.fc(graph_feat)
+    out = self.out(graph_feat)
+    return out
+
+
+class GATModel(TorchModel):
+  """Graph Attention Networks (GAT).
+
+  Here is a simple example of code that uses the GATModel with
+  molecules dataset.
+
+  >> import deepchem as dc
+  >> featurizer = dc.feat.MolGraphConvFeaturizer()
+  >> dataset_config = {"reload": False, "featurizer": featurizer, "transformers": []}
+  >> tasks, datasets, transformers = dc.molnet.load_tox21(**dataset_config)
+  >> train, valid, test = datasets
+  >> model = dc.models.GATModel(loss=dc.models.losses.SoftmaxCrossEntropy(), batch_size=32, learning_rate=0.001)
+  >> model.fit(train, nb_epoch=50)
+
+  This model takes arbitary graphs as an input, and predict graph properties. This model is
+  one of variants of Graph Convolutional Networks. The main difference between basic GCN models
+  is how to update node representations. The GAT uses multi head attention mechanisms which
+  outbroke in NLP like Transformer when updating node representations. The most important advantage
+  of this approach is that we can get the interpretability like how the model predict the value
+  or which part of the graph structure is important from attention-weight. Please confirm
+  the detail algorithms from [1]_.
+
+  References
+  ----------
+  .. [1] Veličković, Petar, et al. "Graph attention networks." arXiv preprint
+     arXiv:1710.10903 (2017).
+
+  Notes
+  -----
+  This class requires PyTorch Geometric to be installed.
+  """
+
+  def __init__(self,
+               in_node_dim: int = 38,
+               hidden_node_dim: int = 64,
+               heads: int = 4,
+               dropout_rate: float = 0.0,
```

Comment:
For consistency with other models, this should be just `dropout`.

Reply:
I fixed!